Tasks/task_base.py

```diff
@@ -0,0 +1,68 @@
+# Standard Library
+import abc
+
+class TaskBase():
+    """
+    Responsible for implementing the core logic of the various actions
+    that a drone can take (ex. Movement, Follow, Heal, Decode). Must, at
+    a minimum, implement do(), is_done(), and exit_task().
+    """
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def do(self):
+        """
+        Does one iteration of the logic for this task
+
+        Parameters
+        ----------
+        None (maybe kwargs in the future?)
+
+        Returns:
+        ----------
+        True if the task is done with its goal, and false otherwise
+        (But do not use to query if the task is done, use is_done() instead)
+        """
+        pass
+
+    @abc.abstractmethod
+    def is_done(self):
```

Comment:
This isn't super critical but since this function just returns the value of `self.done` I would suggest looking into making this a property by using `@property`.

Reply:
Ok - makes sense.